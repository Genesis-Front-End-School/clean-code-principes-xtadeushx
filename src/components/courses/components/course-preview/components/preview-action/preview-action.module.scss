.course__preview-action {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 20px;
  margin-top: 10px;

  & span {
    display: flex;
    align-items: center;
    gap: 10px;
    color: white;
    font-size: 30px;
  }
}

.like,
.disLike {
  width: 35px;
  height: 35px;
  color: var(--color-white);

  &:hover {
    color: var(--color-blue-300);
    cursor: pointer;
    transition: all 0.4s ease;
  }
}