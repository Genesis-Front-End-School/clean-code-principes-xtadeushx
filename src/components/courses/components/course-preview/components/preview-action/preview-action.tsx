import { AiOutlineDislike, AiOutlineLike } from 'react-icons/ai';

import styles from './preview-action.module.scss';

const PreviewAction = () => {
  return (
    <div className={styles['course__preview-action']}>
      <span>
        <AiOutlineLike className={styles.like} /> 0
      </span>
      <span>
        <AiOutlineDislike className={styles.disLike} /> 0
      </span>
    </div>
  )
}

export { PreviewAction };
