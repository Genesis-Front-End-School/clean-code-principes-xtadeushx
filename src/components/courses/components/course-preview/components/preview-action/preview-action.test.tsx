import React from 'react';
import { render } from '@testing-library/react';
import { PreviewAction } from './preview-action';

// Mock the react-icons/ai module
jest.mock('react-icons/ai', () => ({
  AiOutlineLike: 'mock-like-icon',
  AiOutlineDislike: 'mock-dislike-icon',
}));

describe('PreviewAction', () => {
  it('should render the like and dislike icons', () => {
    const { container } = render(<PreviewAction />);
    expect(container).toMatchSnapshot();
    expect(container.querySelector('.mock-like-icon')).toBeInTheDocument();
    expect(container.querySelector('.mock-dislike-icon')).toBeInTheDocument();
  });

  it('should render the like and dislike counts', () => {
    const { getByText } = render(<PreviewAction />);
    expect(getByText('0')).toBeInTheDocument();
  });
});
