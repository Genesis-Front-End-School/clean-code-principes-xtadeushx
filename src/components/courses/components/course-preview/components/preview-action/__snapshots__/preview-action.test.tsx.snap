// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewAction should render the like and dislike icons 1`] = `
<div>
  <div>
    <span>
      <mock-like-icon />
       0
    </span>
    <span>
      <mock-dislike-icon />
       0
    </span>
  </div>
</div>
`;
